caluma/caluma_workflow/domain_logic.py

```diff
@@ -528,3 +528,66 @@ def post_resume(work_item, user, context=None):
         )
 
         return work_item
+
+
+class RedoWorkItemLogic:
+    @classmethod
+    def validate_for_redo(cls, work_item):
+        if work_item.status == models.WorkItem.STATUS_READY:
+            raise ValidationError("Ready work items can't be redone.")
+        ready_work_items = cls._find_ready_in_work_item_tree(
+            work_item, allowed_states=[models.WorkItem.STATUS_READY]
+        )
+        allowed = False
+        for wi in ready_work_items:
+            if work_item.task in wi.get_redoable():
+                allowed = True
```

Comment:
You don't need the `allowed` variable. Just return early in case we are allowed to execute the `redo`. Then the  `ValidationError` below can be thrown without `if`

Reply:
True, done.